apps/server/src/modules/tool/controller/dto/request/oauth2-tool-config.params.ts

```diff
@@ -0,0 +1,40 @@
+import { ExternalToolConfigCreateParams } from '@src/modules/tool/controller/dto/request/external-tool-config.params';
+import { ApiProperty, ApiPropertyOptional } from '@nestjs/swagger';
+import { IsArray, IsBoolean, IsEnum, IsOptional, IsString } from 'class-validator';
+import { ToolConfigType } from '@src/modules/tool/interface/tool-config-type.enum';
+
+export class Oauth2ToolConfigParams extends ExternalToolConfigCreateParams {
```

Comment:
Why not extends from BasicToolConfigParams

Reply:
see above